service/searchAllSongs.js

```diff
@@ -0,0 +1,188 @@
+            {
+                    addToFinalList(notRelevant,increaseListSize);
+            }
+            optionMenu += 1;
+
+        }
+
+    }
+
+    /*This line will read from the local file system, a retrieve the search results
+    to make the search result work for a play you only have to change the .json file directory
+    */
+    // for some reson this didn't work for me  __dirname + '/../data/music.1.json'
+    //MAKE SURE TO CHANGE THE DIRECTORY BELOW!!!!!!!!!!!!!!!!!!!
+    fs.readFile('D:\\CSULB\\presemt\\327\\GitClone\\MusicPlayer\\data\\music.json', (err, rawData) =>
+    {
+
+        if(err) console.log(err);
+        else {
+            let data = JSON.parse(rawData);
+
+
+            /*For each song record this loop Will
+            look at the artist name and song title. When a
+            match is found the songID will be added to an array
+            since the songID is an unique attribute.
+            A list of the top 5 matches will be displayed to
+            the user. The ranking of results will depend on
+            the top 5 hotttnesss of the song.
+            */
+            data.forEach(element => {
+                /*
+                There is no rule stating that an artist and song Title
+                cannot have the same name, so to avoid adding duplication
+                we use if , else if.
+    +            */
+
+                let wordsInArtistName = element.artist.name.split(" ");
+                let wordsInSongTitle = element.song.title.split(" ");
+
+                if((userInput.length == 1) && (userInput.toLowerCase() == element.artist.name.slice(0,1).toLowerCase()) || (userInput == element.song.title.toLowerCase()))
+                {
+                    finalResults.push(element);
+
+                }
+                else if((userInput.toLowerCase() == element.song.title.toLowerCase()) || (userInput.toLowerCase() == element.artist.name.toLowerCase()))
+                {
+                    finalResults.push(element);
+                }
+                else if(wordsInArtistName[0].toLowerCase() == userInput.toLowerCase() || wordsInSongTitle[0].toLowerCase() == userInput.toLowerCase())
+                {
+                    firstWordMatch.push(element);
+                }
+                else if(wordsInArtistName.some(compareInputToRecords) || wordsInSongTitle.some(compareInputToRecords))
+                {
+                    someWordsMatch.push(element);
+                }
+                else if((element.artist.name.toLowerCase().includes(userInput.toLowerCase()) || (element.song.title.toLowerCase().includes(userInput.toLowerCase()))))
+                {
+                    substringMatch.push(element);
+                }
+                else
+                {
+                    notRelevant.push(element);
+                }
+            })
+        }
+
+
+
+            sortList(finalResults);
+            if(finalResults.length > 5)
+            {
```

Comment:
Remove Console.log() lines, make sure your function returns the songs and associated artists instead of printing them out